postcodelookup/postcodelookup-1.0.0/org.wso2.carbon.connector/src/main/resources/postCodeLookup/postcodelookup.xml

```diff
@@ -0,0 +1,96 @@
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied. See the License for the
+   specific language governing permissions and limitations
+   under the License.
+-->
+<template xmlns="http://ws.apache.org/ns/synapse" name="postcodelookup">
+    <parameter name="postCode" description="Post code being searched for"/>
+    <parameter name="responseType" description="Response type being searched for"/>
+    <parameter name="lines" description="Lines being searched for"/>
+    <parameter name="include" description="Include being searched for"/>
+    <parameter name="exclude" description="Exclude being searched for"/>
+    <parameter name="format" description="Format being searched for"/>
+    <parameter name="responseType" description="Response type being searched for"/>
+    <parameter name="addtags" description="Add tags being searched for"/>
+    <parameter name="identifier" description="Identifier being searched for"/>
+    <parameter name="callback" description="Callback being searched for"/>
+    <parameter name="page" description="Page being searched for"/>
+    <parameter name="postcodeonly" description="Post code only or not boolean vlaue"/>
+    <sequence>
+        <property name="uri.var.postCode" expression="$func:postCode"/>
+        <property name="uri.var.responseType" expression="$func:responseType"/>
+        <property name="uri.var.lines" expression="$func:lines"/>
+        <property name="uri.var.include" expression="$func:include"/>
+        <property name="uri.var.exclude" expression="$func:exclude"/>
+        <property name="uri.var.format" expression="$func:format"/>
+        <property name="uri.var.addtags" expression="$func:addtags"/>
+        <property name="uri.var.identifier" expression="$func:identifier"/>
+        <property name="uri.var.callback" expression="$func:callback"/>
+        <property name="uri.var.page" expression="$func:page"/>
+        <property name="uri.var.postcodeonly" expression="$func:postcodeonly"/>
+        <script language="js"><![CDATA[
+            var responseType=mc.getProperty('uri.var.responseType');
+            var lines=mc.getProperty('uri.var.lines');
+            var include=mc.getProperty('uri.var.include');
+            var exclude=mc.getProperty('uri.var.exclude');
+            var format=mc.getProperty('uri.var.format');
+            var addtags=mc.getProperty('uri.var.addtags');
+            var identifier=mc.getProperty('uri.var.identifier');
+            var callback=mc.getProperty('uri.var.callback');
+            var page=mc.getProperty('uri.var.page');
+            var postcodeonly=mc.getProperty('uri.var.postcodeonly');
+            var query="?";
+
+            if(responseType != null && responseType != ""){
```

Comment:
As a suggestion, could we improve this further, concise it by adding all the elements to an array at the initial stage and do the string concatenation through a for-each loop ? , The reason is if the API post code lookup service keeps changing by adding/ omitting query params, having a lot of conditional statements would be hard to maintain.